main-settings/src/main/scala/sbt/Append.scala

```diff
@@ -8,109 +8,96 @@
 package sbt
 
 import java.io.File
-import Def.Classpath
-import scala.annotation.implicitNotFound
-import sbt.internal.util.Attributed
-import Def.Initialize
-import reflect.internal.annotations.compileTimeOnly
+
+import scala.annotation.{ compileTimeOnly, implicitNotFound }
+
+import sbt.Def.{ Classpath, Initialize }
 import sbt.internal.io.Source
+import sbt.internal.util.Attributed
 import sbt.io.{ AllPassFilter, NothingFilter }
 
 object Append {
-  @implicitNotFound(
-    msg = "No implicit for Append.Value[${A}, ${B}] found,\n  so ${B} cannot be appended to ${A}"
-  )
+  @implicitNotFound("No Append.Value[${A}, ${B}] found, so ${B} cannot be appended to ${A}")
   trait Value[A, B] {
     def appendValue(a: A, b: B): A
   }
-  @implicitNotFound(
-    msg = "No implicit for Append.Values[${A}, ${B}] found,\n  so ${B} cannot be appended to ${A}"
-  )
+
+  @implicitNotFound("No Append.Values[${A}, ${B}] found, so ${B} cannot be appended to ${A}")
   trait Values[A, -B] {
     def appendValues(a: A, b: B): A
   }
+
   trait Sequence[A, -B, T] extends Value[A, T] with Values[A, B]
 
   implicit def appendSeq[T, V <: T]: Sequence[Seq[T], Seq[V], V] =
     new Sequence[Seq[T], Seq[V], V] {
       def appendValues(a: Seq[T], b: Seq[V]): Seq[T] = a ++ b
       def appendValue(a: Seq[T], b: V): Seq[T] = a :+ b
     }
+
   implicit def appendSeqImplicit[T, V](implicit ev: V => T): Sequence[Seq[T], Seq[V], V] =
     new Sequence[Seq[T], Seq[V], V] {
-      def appendValues(a: Seq[T], b: Seq[V]): Seq[T] =
-        a ++ (b map { x =>
-          (x: T)
-        })
+      def appendValues(a: Seq[T], b: Seq[V]): Seq[T] = a ++ b.map(x => (x: T))
       def appendValue(a: Seq[T], b: V): Seq[T] = a :+ (b: T)
     }
+
   @compileTimeOnly("This can be used in += only.")
   implicit def appendTaskValueSeq[T, V <: T]: Value[Seq[Task[T]], Initialize[Task[V]]] =
-    new Value[Seq[Task[T]], Initialize[Task[V]]] {
-      def appendValue(a: Seq[Task[T]], b: Initialize[Task[V]]): Seq[Task[T]] = ???
-    }
+    (_, _) => ???
```

Comment:
I think this might decrease the readability of the code.
SAM conversions are great for `onClick = _ => ()` kind of lifting, but in this case, my first reaction was confusion.
I'd ask to at least mention SAM conversion in the commit message, and maybe put a comment next to the code.

```scala
    (_, _) => ??? // lifted to Value using SAM conversion
```

Reply:
I think type annotations can help, but I do agree that sometimes SAM does make code less readable.

Reply:
I'm inclined to think it's only confusing because it's a new thing we didn't use to have and so we're not used to.  And sbt being 2.12 only (that is: not still cross-building to 2.11) makes it one of the front-runners for 2.12-only features like SAM conversions.

Also, the instance you selected here is also special because it's entangled with one of sbt's rewrite macros.  If you pick some of the others I think they're comprehensible:

```scala
implicit def appendSeqN[T, V <: T]: Values[Seq[T], Seq[V]] = _ ++ _
implicit def appendList1[T, V <: T]: Value[List[T], V]     = _ :+ _
implicit def appendFunction[A, B]: Value[A => A, A => A]   = _.andThen(_)
```

Reply:
When the seeming types do not match up, I am inclined to assume "ok it must be implicits." Maybe eventually we'll rewire our thinking to "it must be implicits || SAM conversion." To me `_ ++ _` is equally head scratching.

In the meantime, having both commit comment and code comments would certainly benefit Ethan, me, and others who might wander into this code.

Reply:
> When the seeming types do not match up, I am inclined to assume "ok it must be implicits." Maybe eventually we'll rewire our thinking to "it must be implicits || SAM conversion." To me `_ ++ _` is equally head scratching.

It's the "seeming types do not match up" that'll change, with time.  You're totally cool with brain-compiling `val f: (String, String) => String = _ + _`.  With SAM conversions you'll have just expand that to also brain-compile `val f: Value[String, String] = _ + _`

Reply:
ok. I'm fine with the change, but since it might take a while for my old brain to rebuild plz add the comments etc.

Reply:
> plz add the comments etc.

Will do.